lib/createStylelintResult.js

```diff
@@ -2,36 +2,109 @@
-    ? undefined
-    : postcssResult.root.source.input.file ||
-      postcssResult.root.source.input.id;
+/*:: type deprecationMessageType = {
+  type: string,
+  text: string,
+  stylelintType: string,
+  stylelintReference: string,
+}
+*/
+
+/*:: type invalidOptionMessageType = {
+  type: string,
+  text: string,
+  stylelintType: string,
+}
+*/
+
+/*:: type messageType = {
+  type: string,
+  text: string,
+  line: number,
+  column: number,
+  severity: string,
+  rule: string,
+  node: Object,
+  index: number,
+}
+*/
+
+/*:: type postcssResultType = {
+  processor: {
+    version: string,
+    plugins: Array<any>,
+  },
+  messages: Array<messageType>,
+  root: {
+    raws: {
+      semicolon: boolean,
+      after: string,
+    },
+    type: string,
+    nodes: Array<Object>,
+    source: {
+      input: Object,
+      start: Object,
+    },
+    lastEach: number,
+    indexes: Object,
+  },
+  opts: {
+    from: ?string,
+    syntax: ?{
+      parse: Function,
+      stringify: Function,
+    },
+  },
+  css: string,
+  map: ?any,
+  stylelint: {
+    ruleSeverities: Object,
+    customMessages: Object,
+    quiet: ?any,
+    disabledRanges: {
+      all: Array<any>,
+    },
+    ignored?: boolean,
+    stylelintError?: boolean,
+  },
+}
+*/
+
+/*:: type resultType = {
+  config: {
+    codeProcessors?: Array<Function>,
+    ignorePatterns?: string,
+    processors?: Array<any>,
+    quiet?: boolean,
+    resultProcessors?: Array<Function>,
+    rules: Object,
+  },
+  filepath: string,
+}
+*/
+
+module.exports = function (
+  stylelint/*: stylelint$internalApi*/,
+  postcssResult/*: postcssResultType*/,
+  filePath/*:: ?: string*/
+)/*: Promise<stylelint$result>*/ {
+  const source = !postcssResult.root.source ? undefined : postcssResult.root.source.input.file || postcssResult.root.source.input.id
 
   // Strip out deprecation warnings from the messages
-  const deprecationMessages = _.remove(postcssResult.messages, {
-    stylelintType: "deprecation"
-  });
-  const deprecations = deprecationMessages.map(deprecationMessage => {
+  const deprecationMessages = _.remove(postcssResult.messages, { stylelintType: "deprecation" })
+  const deprecations = deprecationMessages.map((deprecationMessage/*: deprecationMessageType*/) => {
     return {
       text: deprecationMessage.text,
       reference: deprecationMessage.stylelintReference
     };
   });
 
   // Also strip out invalid options
-  const invalidOptionMessages = _.remove(postcssResult.messages, {
-    stylelintType: "invalidOption"
-  });
-  const invalidOptionWarnings = invalidOptionMessages.map(
-    invalidOptionMessage => {
-      return {
-        text: invalidOptionMessage.text
-      };
+  const invalidOptionMessages = _.remove(postcssResult.messages, { stylelintType: "invalidOption" })
+  const invalidOptionWarnings = invalidOptionMessages.map((invalidOptionMessage/*: invalidOptionMessageType*/) => {
+    return {
+      text: invalidOptionMessage.text,
     }
   );
 
```

Comment:
There's a closing `}` missing from the above in  `const invalidOptionWarnings`

When fixing it myself the _flow_ errors throw me off what's going on, I need to read more _flow_ docs

Reply:
The errors are occurring because Flow doesn't understand that lodash is filtering out the different messageTypes, so it doesn't know that a deprecationMessage is a different type than a message. I suggest creating a single unified messageType with optional properties:

```js
/*:: type messageType = {
  type: string,
  text: string,
  line: number,
  column: number,
  severity: string,
  rule: string,
  node: Object,
  index: number,
  stylelintType?: string,
  stylelintReference?: string,
}
*/
```

Reply:
@ivanzusko Is this something you can incorporate?